### ISSUE TITLE
Add handling of "-z" compiler option to ignore it explictly?

### ISSUE THREAD

**Issue by Delgan** [ccache/ccache#1061]
### How to reproduce ###

1. Compile a program with CCache+GCC with `-z defs` option

### Actual behavior ###

Message in logs: `"defs is not a regular file, not considering as input file"`.

### Expected behavior ###

I see there exist a list of options taking an arguments to be explicitly ignored: https://github.com/ccache/ccache/blob/346c94602e85433dfb587f93aee10f2a75c02621/ccache.c#L1550-L1571

I was wondering if this list was meant to be somewhat exhaustive? Should we add the `-z` option thus avoiding a "warning" in the logs and also avoiding to even consider the argument as a file?

The `-z` option accepts a `keyword` and is directly passed to the linker.  For example, the list of valid keyword for `ld` can be found [here](http://manpages.ubuntu.com/manpages/trusty/man1/ld.gold.1.html) and for `gold` they are [here](http://manpages.ubuntu.com/manpages/trusty/man1/ld.gold.1.html).

### Environment ###

ccache version 4.6
Ubuntu 18.04

**Comment by jrosdahl** [comment]
> Message in logs: "defs is not a regular file, not considering as input file".

That message is just a note (or debug message if you will), not a warning. Do you think it's a problem? If so, why?

> I was wondering if this list was meant to be somewhat exhaustive?

No, since keeping up with all compiler versions' new options would be a huge and never-ending task. We only need to describe those that make a difference to ccache's purposes.

**Comment by Delgan** [comment]
It's not a problem per se, as I learned it's just a debug message anyway. :)

When I looked at the logs for the first time, I was a bit surprised by this message and assumed that CCache was maybe misinterpreting one of the arguments, possibly causing the unexpected behavior I was investigating. Then I looked at the code and realized that it was no big deal.

I was thinking that maybe, if `-z` is a known option, it can be added to the CCache white list (avoiding both the message and the IO check).  If this is not desirable for other reasons, and it is better to do the check explicitly, then do not hesitate to close this ticket. :)
